core/src/main/java/io/jenkins/servlet/ServletContextWrapper.java

```diff
@@ -0,0 +1,692 @@
+
+import io.jenkins.servlet.descriptor.JspConfigDescriptorWrapper;
+import java.io.InputStream;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.Collections;
+import java.util.Enumeration;
+import java.util.EventListener;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import java.util.stream.Collectors;
+import javax.servlet.Filter;
+import javax.servlet.FilterRegistration;
+import javax.servlet.RequestDispatcher;
+import javax.servlet.Servlet;
+import javax.servlet.ServletContext;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRegistration;
+import javax.servlet.SessionCookieConfig;
+import javax.servlet.SessionTrackingMode;
+import javax.servlet.descriptor.JspConfigDescriptor;
+
+public class ServletContextWrapper {
+    public static jakarta.servlet.ServletContext toJakartaServletContext(ServletContext from) {
+        Objects.requireNonNull(from);
+        return new jakarta.servlet.ServletContext() {
+            @Override
+            public String getContextPath() {
+                return from.getContextPath();
+            }
+
+            @Override
+            public jakarta.servlet.ServletContext getContext(String uripath) {
+                return toJakartaServletContext(from.getContext(uripath));
+            }
+
+            @Override
+            public int getMajorVersion() {
+                return from.getMajorVersion();
+            }
+
+            @Override
+            public int getMinorVersion() {
+                return from.getMinorVersion();
+            }
+
+            @Override
+            public int getEffectiveMajorVersion() {
+                return from.getEffectiveMajorVersion();
+            }
+
+            @Override
+            public int getEffectiveMinorVersion() {
+                return from.getEffectiveMinorVersion();
+            }
+
+            @Override
+            public String getMimeType(String file) {
+                return from.getMimeType(file);
+            }
+
+            @Override
+            public Set<String> getResourcePaths(String paths) {
+                return from.getResourcePaths(paths);
+            }
+
+            @Override
+            public URL getResource(String path) throws MalformedURLException {
+                return from.getResource(path);
+            }
+
+            @Override
+            public InputStream getResourceAsStream(String path) {
+                return from.getResourceAsStream(path);
+            }
+
+            @Override
+            public jakarta.servlet.RequestDispatcher getRequestDispatcher(String path) {
+                return RequestDispatcherWrapper.toJakartaRequestDispatcher(from.getRequestDispatcher(path));
+            }
+
+            @Override
+            public jakarta.servlet.RequestDispatcher getNamedDispatcher(String path) {
+                return RequestDispatcherWrapper.toJakartaRequestDispatcher(from.getNamedDispatcher(path));
```

Comment:
The method is documented to potentially return `null` so this should account for that.

Reply:
Fixed in the latest revision.